SUITasks/SUITasks/Flow/Countries/CountriesView.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Countries.swift
+//  SUITasks
+//
+//  Created by user on 13.06.2024.
+//
+
+import SwiftUI
+
+struct CountriesView: View {
+    @ObservedObject private var viewModel = CountriesViewModel()
+    @State private var previousCountries: String?
+    @State private var didLoad = false
+    
+    var body: some View {
+        ZStack {
+            List {
+                ForEach(viewModel.countries) { country in
+                    NavigationLink {
+                        DetailCountryView(countryModel: country)
+                    } label: {
+                        CountryCellView(countryModel: country)
+                            .onAppear {
+                                if country == viewModel.countries.last {
+                                    guard let nextCountries = viewModel.nextCountries,
+                                          previousCountries != nextCountries
+                                    else {
+                                        return
+                                    }
+                                    viewModel.fetchCountries(nextCountries)
+                                    previousCountries = nextCountries
+                                }
+                            }
```

Comment:
is it possible to move these functions into a separate functions inside of this view? So that you separate logic from UI